ENCD-4855 Extra-wide search facets

forresttanaka opened the pull request:
I rearranged the matrix views (experiment and audit) so that all views that include facets follow a pattern of a wrapper div with display:flex, and then exactly two child divs: one with the facets taking up a horizontal portion and one with the main content of the page taking the rest. The matrix views now split the search terms input field off from the facets, and the view buttons off the main matrix view, and combines them both into the header div. That leaves just the facets and header content to share horizontal space, matching other pages that have facets.

The cart view display does have facets, and does follow this pattern, but uses different CSS classes and values for a different look, so it doesn’t _quite_ follow the pattern of the others.

I moved `<MatrixInternalTags>` from objectutils.js to matrix.js since it’s a matrix-specific component, and rewrote it to take advantage of React 16’s ability to render arrays of components.

I removed the onFilter function properties passed to FacetList as they aren’t used, and the functions passed to them did nothing beyond what a regular link does — navigate to the href. I believe the use of onFilter can be removed from search.js as well, as it only served the tabbed biodalliance view we don’t have anymore, but I didn’t do that in this ticket.

forresttanaka commented on src/encoded/static/components/auditmatrix.js:
```diff
@@ -375,10 +371,31 @@ const MatrixHeader = ({ context }) => {
+                </div>
+                <div className="matrix-header__search-controls">
+                    <h4>Showing {context.total} results</h4>
+                    <SearchControls context={context} />
```
Dang, both the /matrix/ and /audit/ matrices use `context.matrix.doc_count` to decide whether to issue an error or not, but with new search this property doesn’t exist anymore. I’ll write a new ticket for that, and maybe @hitz will want to hotfix rc2, or wait for v92.

emro commented on src/encoded/static/components/matrix.js:
```diff
@@ -369,24 +369,58 @@ const convertExperimentToDataTable = (context, getRowCategories, getRowSubCatego
+MatrixInternalTags.propTypes = {
+    /** encode search-results object being displayed */
+    context: PropTypes.object.isRequired,
+};
```
One more thought about this: I just remembered I have a ticket to add Internal Tags to search pages, ENCD-4219, so we'll probably actually want this to be called `InternalTags` and move it back to 'objectutils.js'. That can be changed as part of that ticket though. If it were just going to be on the matrix pages this would make the most sense.

forresttanaka commented:
I moved InternalTags back to objectutils.js, and renamed it back to MatrixInternalTags because InternalTags already exists. We might want to try to consolidate those if it’s practical in the future. I also returned the Visualize button to /audit/, and corrected the merge error that allowed visualizations beyond 500 experiments.

forresttanaka commented:
Resolved merge conflicts.
